va/va.h

```diff
@@ -893,6 +908,46 @@ typedef enum
      *  this setting also could be update by \c VAContextParameterUpdateBuffer
      */
     VAConfigAttribContextPriority       = 41,
+    /** \brief TEE could be any HW secure device. Read-only */
+    VAConfigAttribTEEType               = 42,
+    /** \brief TEE type client is a specific module supporting specific functions in TEE. Read-only*/
+    VAConfigAttribTEETypeClient         = 43,
+    /**
+     * \brief Cipher algorithm of the protected content session.
+     *
+     * This attribute specifies the cipher algorithm of the protected content session. It
+     * could be AES, etc....
+     */
+    VAConfigAttribProtectedContentCipherAlgorithm = 44,
+    /**
+     * \brief Cipher block size of the protected content session.
+     *
+     * This attribute specifies the block size of the protected content session. It could be
+     * 128, 192, or 256.
+     */
+    VAConfigAttribProtectedContentCipherBlockSize = 45,
+    /**
+     * \brief Cipher mode of the protected content session.
+     *
+     * This attribute specifies the cipher mode of the protected content session. It could
+     * be CBC, CTR, etc...
+     */
+    VAConfigAttribProtectedContentCipherMode = 46,
+    /**
+     * \brief Decryption sample type of the protected content session.
+     *
+     * This attribute specifies the decryption sample type of the protected content session.
+     * It could be fullsample or subsample.
+     */
+    VAConfigAttribProtectedContentCipherSampleType = 47,
```

Comment:
No. Please refer to [ ISO CENC v2/v3 ](https://www.iso.org/obp/ui/#iso:std:iso-iec:23001:-7:ed-2:v1:en) specs for more details.

Reply:
so, it is not possible to add just one attributes, such as VAConfigAttribProtectedContentCipher
the value is 
typedef union _VAConfigAttribValProtectedContentCipher {
    struct {
     uint32 Algorithm : 8;
     uint32 sample_type :8;
     uint32 mode : 8;
     uint32 reserved: 8;
    } bits;
   uint32 value;
    
}VAConfigAttribValProtectedContentCipher

Reply:
?

Reply:
I thought question was - Could algorithm itself represent other attributes? For which answer is No since HWDRM follows CENC v2/3 specs and those specs define these attributes separately.
Right now, API definition is based on DRM specs(PlayReady or WideVine or FairPlay) and none of those spec combines these values but technically we could do as suggested here and as impact SW has to do translation.
The main reason DRM specs kept these fields separately to allow combinations. For example, if algorithm is AES-CTR then sub and full sample could be sample type and for AES-CBC then sub sample(not full sample) could be sample type. We could implement suggestion but number of calls will increase and perhaps we have to talk to DRM spec implementer's to change their implementation which could be done but I doubt DRM spec implementers will change their implementation for Intel HW.

Reply:
> I thought question was - Could algorithm itself represent other attributes? For which answer is No since HWDRM follows CENC v2/3 specs and those specs define these attributes separately.

no , my question is: seems there are few mode/algorithm/sample_type, so maybe 8 bits is enough for each attributes. so , maybe we could use one uint32 attributes (different bit fields) to represent the attributes. then there are only one attribute. 

> Right now, API definition is based on DRM specs(PlayReady or WideVine or FairPlay) and none of those spec combines these values but technically we could do as suggested here and as impact SW has to do translation.

maybe my expression is not accurate , seems , we dont need 3 attributes, just one attribute to represent 3 attributes, the bits length is enough. 
 
> The main reason DRM specs kept these fields separately to allow combinations. For example, if algorithm is AES-CTR then sub and full sample could be sample type and for AES-CBC then sub sample(not full sample) could be sample type. We could implement suggestion but number of calls will increase and perhaps we have to talk to DRM spec implementer's to change their implementation which could be done but I doubt DRM spec implementers will change their implementation for Intel HW.

Reply:
This is nice idea but this design is bound to current HWDRM implementation and moreover design will not be scalable beyond HWDRM.
Right now in structure, we have Algorithm - AES CTR/CBC, Sample - Full/Sub, Mode - 128/192/256. 
This suites HWDRM implementation where sample construct has single attribute. For example, in HWDRM sample construct is bit-stream and the only attribute [HWDRM sees is 1 sample irrespective of frame boundary, headers etc.](https://www.webmproject.org/vp9/mp4/)  
Tomorrow, we can have sample encryption which is multi-flavor for different attributes of sample construct. For example, 
tomorrow HWDRM could be designed in such a way that single bit-stream construct is divided into following attributes - headers, frame-data, super frame-data etc. and headers could be sub-sample encrypted while frame-data full-sample then above scheme will no scale as each field would be bound as single value.